pkg/lightning/backend/external/writer.go

```diff
@@ -68,6 +69,37 @@ const (
 	DefaultBlockSize = 16 * units.MiB
 )
 
+// GetAdjustedMergeSortOverlapThreshold adjusts the merge sort overlap threshold based on concurrency.
+// The bigger the threshold, the bigger the statistical bias. In CPU:Memory = 1:2 machine, if the concurrency
+// is less than 8, the memory can be used to load data is small, and may get blocked by the memory limiter.
+// So we lower the threshold here if concurrency too low.
+func GetAdjustedMergeSortOverlapThreshold(concurrency int) int64 {
+	intest.Assert(concurrency > 0, "concurrency must be greater than 0, got %d", concurrency)
+	if concurrency > 16 {
+		return maxMergeSortOverlapThreshold
+	}
+	if concurrency < 0 {
```

Comment:
```suggestion
	if concurrency <= 0 {
```
for prod code, i think it's not possible